### DISCUSSION TITLE
Licensing your GitHub repository

### DISCUSSION THREAD

**Discussion by mariussdl** [community/community#120606] [New to GitHub]
Hey,

I have question about licensing my repository. 

I would like to know if it is possible to change the licence of my repo, after having a licence (currently Apache 2.0).
Also, I would like to create my own licence, is that possible ? It would still be an "open-source friendly" licence (of course), but with some more specifications and a few more restrictions.

Thanks for your answers :)

Best,
therealmarius

**Comment by airtower-luna** [comment]
At the *technical* level, you can change the license file like any other file.

Whether you have the *legal right* to change the license is a very different question. :warning: Highly simplified (I'm not a lawyer): Do you alone own the rights to the code? If not, have the other rights holders allowed you to do it? And of course if you write your own license there's the question if it's legally sound (Is it enforceable? Would it do what you want it to?).

And as matter of opinion: If it's someone else's code *and* the license allows you to relicense (some open source licenses do, some don't) changing it to a *not* open source one is still very rude.

**Reply by mariussdl** [reply]
It will still be open source. I simply want to add more terms regarding commercial uses and trademark uses.

Regarding the code owning, can you tell me what you mean by "owning the code" ? I forked a repository with no license and the only thing that I kept is the idea. Other users also opened pull request that were merged. Does that means that they own part of the code ?

Thanks,
therealmarius

**Reply by airtower-luna** [reply]
> I forked a repository with no license and the only thing that I kept is the idea.

"No license" usually means "you're allowed to look and that's it". I'm not going wade into the quagmire of trying guess if what you made is really completely separate. :sweat_smile: 

> Other users also opened pull request that were merged. Does that means that they own part of the code ?

Very likely, but copyright law is complicated (not everything is copyrightable, and it varies by jurisdiction). The other question is under which license they made their contributions. Rule of thumb: If you have a code base with many contributors and they *didn't* give up their rights through a CLA or similar, changing the license is practically impossible unless the previous license allows relicensing: You'd either need to get *everyone* to agree, or figure out how to pry it apart and remove everything contributed by people who don't agree.

And if the previous license allows relicensing you'd still need to follow other requirements (e.g. attribution for BSD-style licenses).
